### PR TITLE
Exclude bad times from validation

### PR DESCRIPTION
**Pull request by jeanconn**
This would exclude bad time ranges (specified in the characteristics) completely from validation violations, but leave the values in the plots (with grey background).

### CODE REVIEW

**Comment by jeanconn** [comment]
I thought of using the bad times from eng_archive, but those seemed more specific and these can be both broader and more casually determined for this application.

**Comment by jeanconn** [comment]
obsid_valid

File: validate_states.py [hunk 1]
```diff
@@ -195,6 +196,15 @@ def get_telem_values(tstart, msids, days=14, dt=32.8, name_map={}):
     return out
 
 
+def get_bad_mask(tlm):
+    mask = np.zeros(len(tlm), dtype='bool')
+    for interval in characteristics.bad_times:
+        bad = ((tlm['date'] >= DateTime(interval['start']).secs)
```
Comment (taldcroft) [inline validate_states.py]: Not a big deal here, but in general if you are getting intervals from sorted time lists you should use np.searchsorted like
```
i0, i1 = np.searchsorted(tlm['date'], [tstart, tstop])
mask[i0:i1] = True
```
This is way faster, although I confess that I always have to think about off-by-one issues at the edges.  Even here I'm not 100% sure if what I wrote is exactly right so you can leave this if you want.  But when performance is an issue...

**Comment by taldcroft** [comment]
:+1:

File: validate_states.py (hunk 1 above)
Comment (jeanconn) [inline validate_states.py]: OK.  Thanks.  tlm is a relatively small data set and performance isn't an issue here, so yes, I think I will leave this one, but will be smarter about this for similar searches.

**Pull request merged**